### PR TITLE
Feature: Test branch

### PR DESCRIPTION
**Pull request by luftedar**
For this branch, I've created unit test for all the application components  using :100: 

-  Jest snapshots 
- reacting libraries to simulate user interaction.

Please check theses changes and send me your feedback before to move forward the next steps.
Cheers,

### CODE REVIEW

File: src/__tests__/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  test('AC button test', () => {
+    obj = calculate(obj, 'AC');
+    const nullObj = {
+      total: null,
+      next: null,
+      operation: null,
+    }
+    /*result = calculate(obj,'÷');
+    result = calculate(obj,'5');
+    result = calculate(obj,'=');*/
```
Comment (omar-labana) [inline src/__tests__/calculate.test.js:34]: - [ ] _[OPTIONAL]_ I would suggest testing each case with a standalone object, currently test cases use the same object output from the previous test cases.
- [ ] Please remove the commented code above.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi team, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
- [ ] Please use Jest snapshots testing to cover components with UI tests.
- [ ] Kindly make use of the `React Testing Library` to simulate user interaction.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by taher-software** [comment]
Hi @omar-labana 
Thank you for your feedback.
I didn't understand exactly what I need to change, please can you explain in more details the required changes.
Thank you for your collaboration .

**Comment by omar-labana** [comment]
Hi @Taher-web-dev, 
I apologize for not noticing the existing tests in a different directory. Kindly submit another CRR.

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @Taher-web-dev @luftedar 

Amazing job!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**